### DISCUSSION TITLE
Ho to re-render a q-markup-table

### DISCUSSION THREAD

**Discussion by CynthiaBlue** [quasarframework/quasar#11218] [CLI - PWA mode]
I am using a q-markup-table that I need to have auto-reload when a button is pressed when something is added to the IndexedDB table via Dexie.

I've been trying the methods in this tutorial:
[https://michaelnthiessen.com/force-re-render/](https://michaelnthiessen.com/force-re-render/)
but using the key-changing method isn't re-rendering my q-markup-table.

Maybe I'm not sure where to put the :key inside the table tags.  

I currently have the key inside the q-markup-table tag:
```<q-markup-table flat bordered :separator="separator" :key="componentKey">```

But do I need in the tr or td tag?
or does the method not work with a q-markup-table?

I've tried wrapping the q-markup-table in a div but that's not working either.
```<div :key="componentKey">```

I may or may not change this into a q-table.  I've been having problems getting the data into the q-table so I'm working with the markup table for now.

**Comment by metalsadman** [comment]
be sure `componentKey` is reactive, and you are changing it different from previous assigned value to it.

**Reply by CynthiaBlue** [reply]
> be sure `componentKey` is reactive, and you are changing it different from previous assigned value to it.

Thahks for your reply. I'm still really new to Quasar, how do I make sure 'componentKey' is reactive?

**Comment by rstoenescu** [comment]
Why would you want to re-render? Is your data scope changing or not? If it is, that's the point with Vue -- it takes care of any necessary updates. But regardless, the answer above this post is correct.

**Reply by CynthiaBlue** [reply]
I need to reload a table when I add new data to the database. The table reflects the data in the table. Though now I am using a q-table instead of a markup table.  I still can't figure out how to reload the table with refreshed data after I have added or deleted data to/from the database.

**Reply by metalsadman** [reply]
use a unique reactive `key` and update it to force a re-render ie. `<q-table :key="someKey">` `key.value = Date.now()`, though if you're just trying to update the data, then update the data that you passed on qtable's `rows` prop.
